### PR TITLE
Issue #217 Fix to eliminate re-counting of retried test cases, and append "retry iteration #" on names of retried tests

### PR DESCRIPTION
**Pull request by ssharma0312**
Fix for https://github.com/XCTestHTMLReport/XCTestHTMLReport/issues/217
#217 

Fix will include:
1) Eliminating duplicate counts of retried tests in HTML as well JUNIT report
image
All = Total Unique Tests (Same as XCode)
Passed = Passed + Mixed (These tests may have failed but passed on retry, they are categorized as Mixed in XCode)
Failed = Total Unique Failed Tests (It won't count retried tests)

As now you can see we match closely with XCode pattern.

2) Appending "retry iteration #" to different retried tests on HTML report
image

### CODE REVIEW

File: prepareTestResults.sh [hunk 1]
```diff
@@ -11,8 +11,15 @@ xcodebuild test \
...
+#    -test-iterations 5 \
+#    -retry-tests-on-failure \
+#    -resultBundlePath "$FILENAME" || true
 echo "Even if some test failed this is OK."
 echo "${FILENAME} should contain succeed, failed and skipped tests for xchtmlreport functional testing"
 rm -rf "../Tests/XCTestHTMLReportTests/${FILENAME}"
 mv "$FILENAME" "../Tests/XCTestHTMLReportTests/"
-echo "$(tput setaf 2)$(basename "$0") successfully finished$(tput sgr 0)"
\ No newline at end of file
+echo "$(tput setaf 2)$(basename "$0") successfully finished$(tput sgr 0)"
```
Comment (tylervick) [inline prepareTestResults.sh]: I think this should be reverted :)

**Review by tylervick — COMMENTED** [review]
Thank you for this @ssharma0312! Would you mind updating the test sample generation & test scripts to cover this as well?

I'm also wondering if these changes need to be upstream in https://github.com/davidahouse/xcresultkit

File: prepareTestResults.sh (hunk 1 above)
Comment (ssharma7atb) [reply prepareTestResults.sh]: @tylervick - If you just pass flags `-retry-tests-on-failure` and `test-iterations 5`, then XCode 13 will re-run failed test until they pass for a max of 5 times. I have updated the `prepareTestResults` script file to reflect same.

**Comment by ssharma0312** [comment]
@tylervick @nacho4d - Need some reviewers please on this PR? What are the next steps before this can be merged to master, and included in next release?

**Pull request closed without merging**